TCAPracticeCore/Sources/TCAPracticeCore/QuizFeatureReducer.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  QuizFeatureReducer.swift
+//  TCAPractice
+//
+//  Created by S J on 12/22/23.
+//
+
+//keep business logic away from views, that's why I put it in TCAPracticeCore(platform independent)
+import Foundation
+import ComposableArchitecture
+
+public enum SwipeType {
+  case left
+  case right
+}
+
+@Reducer
+public struct QuizFeatureReducer {
+
+  public init(){}
+
+  public struct State: Equatable {
+
+    public init(allQuestions: [Question] = [], currentQuestionViewModel: QuestionViewModel? = nil, currentQuestion: Question? = nil, offset: CGFloat = 0) {
+      self.allQuestions = allQuestions
+      self.currentQuestionViewModel = currentQuestionViewModel
+      self.currentQuestion = currentQuestion
+      self.offset = offset
+    }
+    
+    public var allQuestions: [Question]  = []// mutate it and remove answered correct question from it
+    public var currentQuestionViewModel: QuestionViewModel?
+    public var currentQuestion: Question?
+    public var offset: CGFloat = 0
+  }
+  
+  public enum Action: Equatable {
+    case loadAllQuestions
+    case nextQuestion(_ previousQuestion: Question? = nil)
```

Comment:
It doesn't look like `previousQuestion` is used here. Do you have future plans for it? Otherwise consider deleting the field.

Reply:
It was actually used in L78

Reply:
It's not used by the action handler though.